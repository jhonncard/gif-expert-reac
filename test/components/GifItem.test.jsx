import {render, screen} from '@testing-library/react';
import {GifItem}from '../../src/components/GifItem';

describe('Name of de group', () => {
  const title ='saitama';
  const url = 'https://jac.com/saitama.jpg';

  test('debe hacer match con el snapshot', () => {
    const {container} =  render( <GifItem title={title} url={url} /> );
    expect(container).toMatchSnapshot();
  });

  test(' debe de mostrat la imagen y el url y el alt indicado', () => {
    render( <GifItem title={ title }  url={ url } /> );
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
 
  test('debe de mostrta el titulo en el componente', () => {
    render( <GifItem title={ title }  url={ url } /> );
    expect( screen.getByAltText(title)).toBeTruthy();
  })




});