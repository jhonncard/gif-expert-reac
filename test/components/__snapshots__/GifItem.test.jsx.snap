// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of de group debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://jac.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
